src/backend/distributed/executor/multi_client_executor.c

```diff
@@ -94,6 +94,13 @@ MultiClientConnect(const char *nodeName, uint32 nodePort, const char *nodeDataba
 	char *effectiveDatabaseName = NULL;
 	char *effectiveUserName = NULL;
 
+	if (IsModifyingTransaction)
+	{
+		ereport(ERROR, (errcode(ERRCODE_ACTIVE_SQL_TRANSACTION),
+						errmsg("cannot open new connection during modifying "
```

Comment:
Reworded it, but skipped that `HINT`, since they're supposed to be reserved for providing advice about how to solve a problem.

I suppose a more detailed description could go in the `detail`, but wanted to see if the new message was sufficient before adding that.